libs/utils/env.py

```diff
@@ -439,6 +441,54 @@ def _init_target(self, force = False):
             self.__modules = ['bl', 'cpufreq']
             platform = Platform(model='pixel')
 
+        # Initialize gem5 platform
+        elif self.conf['board'].upper() == 'GEM5':
+            self.nrg_model = None
+            self.__modules=['cpufreq']
+
+            system_info = self.conf['board_settings']['system']
+            simulator_info = self.conf['board_settings']['simulator']
```

Comment:
Yeah the fact that it already has "system" and "simulator" makes it at least specific to models. *Maybe* we could re-use this structure if we were to integrated something like QEMU or ARM Fast Models but TBH our odds are pretty slim of successfully designing a generic interface without investigating the details of those models in this context.